examples/ErrorHandling/HandleExpandedTextAdPolicyViolations.php

```diff
@@ -0,0 +1,245 @@
+        ]);
+
+        // Creates an ad group ad to hold the above ad.
+        $adGroupAd = new AdGroupAd([
+            'ad_group' => $adGroupResourceName,
+            'status' => AdGroupAdStatus::PAUSED,
+            // Sets the expanded text ad info on an Ad.
+            'ad' => new Ad([
+                'expanded_text_ad' => $expandedTextAdInfo,
+                'final_urls' => [new StringValue(['value' => 'http://www.example.com'])]
+            ])
+        ]);
+
+        // Creates an ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+
+        $ignorablePolicyTopics = [];
+        try {
+            // Try sending a mutate request to add the ad group ad.
+            $adGroupAdServiceClient->mutateAdGroupAds($customerId, [$adGroupAdOperation]);
+        } catch (GoogleAdsException $googleAdsException) {
+            $ignorablePolicyTopics = self::fetchIgnorablePolicyTopics($googleAdsException);
+        }
+
+        // Try sending exemption requests for creating an expanded text ad.
+        self::requestExemption(
+            $customerId,
+            $adGroupAdServiceClient,
+            $adGroupAdOperation,
+            $ignorablePolicyTopics
+        );
+    }
+
+    /**
+     * Collects all ignorable policy topics that will be sent for exemption request later.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @return StringValue[] the ignorable policy topics
+     */
+    private static function fetchIgnorablePolicyTopics(GoogleAdsException $googleAdsException)
+    {
+        $ignorablePolicyTopics = [];
+
+        printf("Google Ads failure details:%s", PHP_EOL);
+        foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+            /** @var GoogleAdsError $error */
+            if ($error->getErrorCode()->getErrorCode() !== 'policy_finding_error') {
+                // This example supports sending exemption request for the policy finding error
+                // only.
+                throw $googleAdsException;
+            }
+
+            printf(
+                "\t%s: %s%s",
+                $error->getErrorCode()->getErrorCode(),
+                $error->getMessage(),
+                PHP_EOL
+            );
+            if (!is_null($error->getDetails())
+                && !is_null($error->getDetails()->getPolicyFindingDetails())) {
+                $policyFindingDetails = $error->getDetails()->getPolicyFindingDetails();
+                printf("\tPolicy finding details:%s", PHP_EOL);
+
+                foreach ($policyFindingDetails->getPolicyTopicEntries() as $policyTopicEntry) {
+                    /** @var PolicyTopicEntry $policyTopicEntry */
+                    $ignorablePolicyTopics[] = $policyTopicEntry->getTopic();
```

Comment:
Is there a specific reason for returning an array of StringValues here rather than calling `getTopicUnwrapped()` and returning an array of strings?

Reply:
Because the array of SringValue will be used to request exemption.
It makes no sense to get each as primitive string, just to create a StringValue to wrap it later.